pkg/digest/set_builder.go

```diff
@@ -2,11 +2,15 @@ package digest
 
 import (
 	"sort"
+	"sync"
 )
 
 // SetBuilder is a builder for Set objects.
 type SetBuilder struct {
 	digests map[Digest]struct{}
+
+	// Protects access to digests
+	lock sync.Mutex
```

Comment:
This seems like a pessimisation for the majority of places where SetBuilder is used in synchronous cases.

Reply:
sync.Mutex should be cheap when not contended, but this belongs in the parallelisation PR anyway